src/plugins/console/public/application/components/variables/variables_flyout.tsx

```diff
@@ -0,0 +1,170 @@
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { useState, useCallback, ChangeEvent, FormEvent } from 'react';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n-react';
+
+import {
+  EuiFlyout,
+  EuiFlyoutHeader,
+  EuiTitle,
+  EuiFlyoutBody,
+  EuiFlyoutFooter,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiButton,
+  EuiButtonEmpty,
+  EuiBasicTable,
+  EuiFieldText,
+  useGeneratedHtmlId,
+  EuiForm,
+  EuiButtonIcon,
+  type EuiBasicTableColumn,
+} from '@elastic/eui';
+
+import * as utils from './utils';
+
+export interface DevToolsVariablesFlyoutProps {
+  onClose: () => void;
+  onSaveVariables: (newVariables: DevToolsVariable[]) => void;
+  variables: [];
+}
+
+export interface DevToolsVariable {
+  id: string;
+  name: string;
+  value: string;
+}
+
+export const DevToolsVariablesFlyout = (props: DevToolsVariablesFlyoutProps) => {
+  const [variables, setVariables] = useState<DevToolsVariable[]>(props.variables);
+  const formId = useGeneratedHtmlId({ prefix: '__console' });
+
+  const addNewVariable = useCallback(() => {
+    setVariables((v) => [...v, utils.generateEmptyVariableField()]);
+  }, []);
+
+  const deleteVariable = useCallback(
+    (id: string) => {
+      const updatedVariables = utils.deleteVariable(variables, id);
+      setVariables(updatedVariables);
+    },
+    [variables]
+  );
+
+  const onSubmit = useCallback(
+    (e: FormEvent<HTMLFormElement>) => {
+      e.preventDefault();
+      props.onSaveVariables(variables.filter(({ name, value }) => name.trim() && value));
+    },
+    [props, variables]
+  );
+
+  const onChange = useCallback(
+    (event: ChangeEvent<HTMLInputElement>, id: string) => {
+      const { name, value } = event.target;
+      const editedVariables = utils.editVariable(name, value, id, variables);
+      setVariables(editedVariables);
+    },
+    [variables]
+  );
+
+  const columns: Array<EuiBasicTableColumn<DevToolsVariable>> = [
+    {
+      field: 'name',
+      name: i18n.translate('console.variablesPage.variablesTable.columns.variableHeader', {
+        defaultMessage: 'Variable',
+      }),
+      render: (name, { id }) => {
+        const isInvalid = name && !name.match(/^[a-zA-Z0-9]+$/g);
+        return (
+          <EuiFieldText
+            data-test-subj="variablesNameInput"
+            name="name"
+            value={name}
+            onChange={(e) => onChange(e, id)}
+            isInvalid={isInvalid}
+          />
+        );
+      },
+    },
+    {
+      field: 'value',
+      name: i18n.translate('console.variablesPage.variablesTable.columns.valueHeader', {
+        defaultMessage: 'Value',
+      }),
+      render: (value, { id }) => (
+        <EuiFieldText
```

Comment:
We need to make this input screenreader accessible by adding an `aria-label` prop:

```tsx
        <EuiFieldText
          data-test-subj="variablesValueInput"
          name="value"
          onChange={(e) => onChange(e, id)}
          value={value}
          aria-label={i18n.translate('console.variablesPage.variablesTable.valueInput.ariaLabel', {
            defaultMessage: 'Variable value',
          })}
        />
```